ddtrace/baggage/baggage_test.go

```diff
@@ -182,38 +186,46 @@ func TestBaggageMapAccessorsMakeCopies(t *testing.T) {
 	})
 }
 
-// func TestConcurrentAccess(t *testing.T) {
-// 	ctx := context.Background()
-// 	ctx = Set(ctx, "init", "value")
-
-// 	var wg sync.WaitGroup
-// 	done := make(chan struct{})
-
-// 	// Goroutine 1: Iterates over baggage repeatedly
-// 	wg.Add(1)
-// 	go func() {
-// 		defer wg.Done()
-// 		for {
-// 			select {
-// 			case <-done:
-// 				return
-// 			default:
-// 				_ = All(ctx)
-// 				runtime.Gosched()
-// 			}
-// 		}
-// 	}()
-
-// 	// Goroutine 2: Modifies baggage repeatedly
-// 	wg.Add(1)
-// 	go func() {
-// 		defer wg.Done()
-// 		for range [1000]int{} {
-// 			ctx = Set(ctx, "key", "val")
-// 			runtime.Gosched()
-// 		}
-// 		close(done)
-// 	}()
-
-// 	wg.Wait()
-// }
+// guarantees we also test the Clear→Set path
+func TestConcurrentAccessAndClear(t *testing.T) {
+	base := Set(context.Background(), "init", "val")
+	want := All(base)
+	const readers = 4
+	const writers = 4
+	const iters = 10_000
```

Comment:
Since we are not failing early now (it was a good catch), maybe we can reduce this number.